app/helpers/handlers.js

```diff
@@ -383,22 +373,24 @@ const showPostPage = (currentQuestion, request, h) => {
       }
       break
     case 'project-cost-solar':
-      const calculatedGrantCalfVar = getYarValue(request, 'calculatedGrantCalf')
       // ineligible as calf housing too low
-      if (calculatedGrantCalfVar < 15000) {
+      if (getYarValue(request, 'calculatedGrantCalf') < 15000) {
         return h.view('not-eligible', NOT_ELIGIBLE)
       // calf housing only
-      } else if (calculatedGrantCalfVar >= 500000) {
+      } else if (getYarValue(request, 'calculatedGrantCalf') >= 500000) {
         setYarValue(request, 'calculatedGrant', 500000)
         return h.redirect('/upgrading-calf-housing/potential-amount-conditional')
         // solar capping
       } else if (getYarValue(request, 'calculatedGrant') > 500000) {
-        const newCap = 500000 - getYarValue(request, 'calculatedGrantCalf')
+        const solarCap = 500000 - getYarValue(request, 'calculatedGrantCalf')
         // store capped solar value for potential amount solar capped page
-        setYarValue(request, 'calculatedGrantSolarPreCap', getYarValue(request, 'calculatedGrantSolar'))
+        setYarValue(request, 'calculatedGrantSolarPreCap', getYarValue(request, 'SolarPVCost') * 0.25)
         // set cap for solar and grant
-        setYarValue(request, 'calculatedGrantSolar', newCap)
-        setYarValue(request, 'calculatedGrant', 500000)
+        if (getYarValue(request, 'calculatedGrantSolar') > solarCap) {
+          setYarValue(request, 'calculatedGrantSolar', solarCap)
+          setYarValue(request, 'calculatedGrant', 500000)
+        }
+        setYarValue(request, 'solarCap', solarCap)
```

Comment:
if we are setting calculatedGrantSolar to the solarCap, why do we need a seperate yarKey for solarCap?

Reply:
its in cases when cap is more than grant for solar

Reply:
changed